internal/commands/result.go

```diff
@@ -1040,6 +1040,32 @@ func setIsContainersEnabled(agent string, featureFlagsWrapper wrappers.FeatureFl
 	containerEngineCLIEnabled, _ := wrappers.GetSpecificFeatureFlag(featureFlagsWrapper, wrappers.ContainerEngineCLIEnabled)
 	wrappers.IsContainersEnabled = containerEngineCLIEnabled.Status && agentSupported
 }
+
+func filterResultsByAgent(results *wrappers.ScanResultsCollection, agent string) *wrappers.ScanResultsCollection {
+	if agent == commonParams.VSCodeAgent {
+		var filteredResults []*wrappers.ScanResult
+		for _, result := range results.Results {
+			if result.Type != commonParams.SCSScorecardType {
+				filteredResults = append(filteredResults, result)
+			} else {
+				results.TotalCount--
+			}
+		}
+		results.Results = filteredResults
+	} else if agent != commonParams.DefaultAgent {
```

Comment:
extract the code here to new private method:
filterResultsForDefaultAgent

Reply:
same here filterScsFromResults and ger array of agent